### PR TITLE
Docker improvements

### PR DESCRIPTION
**Pull request by ropc**
- improved `Dockerfile` to make incremental builds faster
- added docker compose configs for local usage/prod usage

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -169,3 +169,5 @@ dmypy.json
 
 # End of https://www.gitignore.io/api/macos,python,visualstudiocode
 env.list
+prod.env
+debug.env
```
Comment (damarti) [inline .gitignore:173]: I just made this `*.env` in mine in case we wanted to do any other ones

File: docker-compose.prod.yml [hunk 2]
```diff
@@ -0,0 +1,7 @@
+version: '3'
+
+services:
+    meme-bot:
+        env_file:
+            - prod.env
```
Comment (damarti) [inline docker-compose.prod.yml:6]: Nice use of this, I actually am just learning about this myself.

File: Dockerfile [hunk 3]
```diff
@@ -2,8 +2,13 @@ FROM python:3.8
...
-COPY fonts/* /usr/share/fonts/truetype/
 WORKDIR /tmp/meme-bot
-RUN poetry install --no-dev
-CMD [ "poetry", "run", "bot" ]
+COPY poetry.lock pyproject.toml ./
+RUN poetry install --no-dev --no-root
+COPY fonts/* /usr/share/fonts/truetype/
+COPY assets ./assets
+COPY meme_generator ./meme_generator
+COPY memebot ./memebot
```
Comment (damarti) [inline Dockerfile:11]: Is there a reason you don't just copy the entire directory? Curious, not particularly convinced one way or the other is right

File: .gitignore (hunk 1 above)
Comment (ropc) [reply .gitignore:173]: yeah i saw that. i'll rebase on top of your changes after yours gets merged and remove this.

File: Dockerfile (hunk 3 above)
Comment (ropc) [reply Dockerfile:11]: just cuz it wasn't necessary? seems like docker wants you to cache as much as possible. so if something changes in the `README.md` for example, there really isn't a need to do the next `RUN poetry install --no-dev --no-interaction`.

that said, looks like i missed tests there =P. so i'll add those

**Pull request merged**